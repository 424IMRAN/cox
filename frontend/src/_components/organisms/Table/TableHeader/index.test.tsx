import React from 'react'
import MUITextField from '@material-ui/core/TextField'
import { mount, shallow } from 'enzyme'
import { HeaderGroup } from '.'

describe('TableHeder  Component', () => {
  it('Renders correctly', () => {
    const wrapper = shallow(
      <HeaderGroup selectedRows={6} indeterminate onClick={null} />
    )
    expect(wrapper).toMatchSnapshot()
  })

  it('Allows us to set props', () => {
    const wrapper = mount(
      <HeaderGroup selectedRows={6} indeterminate onClick={null} />
    )
    expect(wrapper.props().selectedRows).toEqual(6)
    expect(wrapper.props().indeterminate).toEqual(true)
    wrapper.setProps({ selectedRows: 0 })
    wrapper.setProps({ indeterminate: false })
    expect(wrapper.props().selectedRows).toEqual(0)
    expect(wrapper.props().indeterminate).toEqual(false)
  })
})
