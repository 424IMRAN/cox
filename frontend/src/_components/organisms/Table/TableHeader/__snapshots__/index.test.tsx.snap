// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeder  Component Renders correctly 1`] = `
<WithStyles(ForwardRef(Grid))
  alignItems="center"
  className="makeStyles-headerGroup-3"
  container={true}
  direction="row"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <CheckBoxComponent
      customStyle="makeStyles-checkBoxStyle-4 makeStyles-rightSpacing-6"
      handleChange={null}
      indeterminate={true}
      isChecked={false}
    />
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <Typography
      align="left"
      className="makeStyles-selectedText-5 makeStyles-rightSpacing-6"
      variant="body2"
    >
      6 selected
    </Typography>
  </WithStyles(ForwardRef(Grid))>
  <WithStyles(ForwardRef(Grid))
    container={true}
    item={true}
    spacing={3}
    xs={8}
  >
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <ButtonComponent
        className="makeStyles-buttonStyle-7"
        color="#FFFFFF"
        height="28px"
        startIcon="edit"
        textStyle="makeStyles-selectedText-5"
        variant="outlined"
      >
        Edit
      </ButtonComponent>
    </WithStyles(ForwardRef(Grid))>
    <WithStyles(ForwardRef(Grid))
      item={true}
    >
      <ButtonComponent
        className="makeStyles-buttonStyle-7"
        color="#FFFFFF"
        height="28px"
        startIcon="history"
        textStyle="makeStyles-selectedText-5"
        variant="outlined"
      >
        History
      </ButtonComponent>
    </WithStyles(ForwardRef(Grid))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
